pkg/insightstore/chunk.go

```diff
@@ -0,0 +1,280 @@
+// Copyright 2020 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package insightstore
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+// deploy frequency
+
+// DeployFrequencyChunk satisfy the interface Chunk.
+type DeployFrequencyChunk struct {
+	AccumulatedTo int64                    `json:"accumulated_to"`
+	DataPoints    DeployFrequencyDataPoint `json:"data_points"`
+	FilePath      string
+}
+
+type DeployFrequencyDataPoint struct {
+	Daily   []*DeployFrequency `json:"daily"`
+	Weekly  []*DeployFrequency `json:"weekly"`
+	Monthly []*DeployFrequency `json:"monthly"`
+	Yearly  []*DeployFrequency `json:"yearly"`
+}
+
+// DeployFrequency satisfy the interface DataPoint.
+type DeployFrequency struct {
+	Timestamp   int64   `json:"timestamp"`
+	DeployCount float32 `json:"deploy_count"`
+}
+
+func (d *DeployFrequencyChunk) GetFilePath() string {
```

Comment:
nit: `func (c DeployFrequencyChunk) ...`

Reply:
Did You mean use DeployFrequencyChunk instead of *DeployFrequencyChunk?

Reply:
Yes.

Reply:
Only, functions to read some fields?(GetFilePath, GetAccumulatedTo, GetDataPoints, DataCount)

Or, use DeployFrequencyChunk too for functions to set some fields, and, should we return the changed struct?
 
```
func (c ChangeFailureRateChunk) SetFilePath(path string) ChangeFailureRateChunk {
	c.FilePath = path
         return c
}
```

Reply:
Ah sorry, I did not check about the `Set` functions. OK to be simple for this PR let's keep them as pointers at this time. We will consider this thing later.